neo4j-ogm-docs/src/main/asciidoc/reference/nativetypes.adoc

```diff
@@ -0,0 +1,183 @@
+
+To opt-in, please first add the corresponding module for your <<reference:native-property-types:supported-drivers, driver>> and
+than use the new configuration property `use-native-types`:
+
+.Enabling the usage of native types
+[%autowidth.spread,frame="topbot",options="header"]
+|======================
+|ogm.properties   | Java Configuration
+
+a|
+[source, properties]
+----
+URI=bolt://neo4j:password@localhost
+use-native-types=true
+----
+
+a|
+[source, java]
+----
+Configuration configuration = new Configuration.Builder()
+        .uri("bolt://neo4j:password@localhost")
+        .useNativeTypes()
+        .build()
+----
+|======================
+
+The same works with a file based URI for the embedded driver.
+
+Once enabled, native types are used for all attributes of all node- and relationship-entities and also for all parameters passed through the OGM `Session` interface.
+
+[[reference:native-property-types:mapping]]
+== Mapping of native types
+
+The following table describes how Neo4j temporal and spatial property types are mapped to attributes of Neo4j-OGM entities:
+
+======
+[options="header", cols="m,"]
+|===
+| Neo4j type        | Neo4j-OGM type
+| Date | `LocalDate`
+| Time | `OffsetTime`
+| LocalTime | `LocalTime`
+| DateTime | `ZonedDateTime`
+| LocalDateTime | `LocalDateTime`
+| Duration | `TemporalAmount`*
+| Point | One of four different Neo4j-OGM  in `org.neo4j.ogm.types.spatial`**
+|===
+
+
++*+ The Neo4j `Duration` can either be a Java 8 `Duration` or `Period` with the least common denominator being a `TemporalAmount`.
+A Java 8 duration always deals with an exact number of seconds while periods take daylight saving times and others into account when added to instants.
+If you are sure that you only deal with one or the other, you just use an explicit mapping to either `java.time.Duration` or `java.time.Period`.
+
++**+ There is no generic Java type representing a spatial point.
+As OGM supports different ways of connecting to Neo4j it cannot expose either the Java drivers or the internal representation of a point, so it provides a point of it's own.
+Please read the next section to learn which concrete classes Neo4j-OGM offers for a point.
+======
+
+[[reference:native-property-types:spatial-types]]
+== Mapping of Neo4j spatial types
+
+Neo4j supports four slightly different property types for spatial points, see https://neo4j.com/docs/cypher-manual/current/syntax/spatial/[Spatial values].
+All variations of the `Point` type are backed by an index and therefore perform very well in queries.
+The main difference between them is the coordinate reference system.
+A point can either be stored in a geographic coordinate system with longitude and latitude or in a cartesian system with x and y.
+If you add the third dimension, you add altitude or a z-axis.
+
+Geographic coordinate systems are based on a spheroidal surface.
+Coordinates are projections of points on that surface to a flat plane.
```

Comment:
It could be thought of as a projection, but normally you would not say that, but just say the latitude and longitude define the position on the sphere in terms of angles. There are many projections that use WGS84, and one trivial one is to simply treat degrees of latitude and degrees of longitude as equivalent cartesian measures on a plane, which leads to massive distortions as you move from the equator, so this projection is not usually used. Neo4j itself never actually projects this data, except internally in the spatial index, but that should be seen as an implementation detail. We planned to replace the current index projection with one more like googles cube, with much less distortion, but have not done so (yet). There are many options for index projection, and we'd like to keep this internal implementation detail.